ddtrace/_events.py

```diff
@@ -0,0 +1,112 @@
+from typing import Callable
+from typing import Mapping
+from typing import Optional
+from typing import Type
+from typing import Union
+
+import attr
+
+from ddtrace import Span
+from ddtrace import _hooks
+from ddtrace import config
+from ddtrace.internal import compat
+
+
+_HOOKS = _hooks.Hooks()
+
+
+@attr.s(frozen=True)
+class IntegrationEvent(object):
+    """
+    An IntegrationEvent is emitted by an integration (e.g. the flask framework integration)
+    and is linked to a span.
+    """
+    span = attr.ib(type=Span)
+    integration = attr.ib(type=str)
+
+    def emit(self):
```

Comment:
We definitely want typing in all this file :)

Reply:
Sure! I will add more types & unit tests for this file.